application/app/controllers/application_controller.rb

```diff
@@ -15,4 +17,20 @@ def load_user_settings
     Current.settings = UserSettings.new
     @active_project = Project.find(Current.settings.user_settings.active_project.to_s)
   end
+
+  def set_redirect_params
+    Current::PERSISTED_ATTRIBUTES.each do |key|
+      value = request.request_parameters[key] || params[key] || flash[:redirect_params]&.[](key.to_s)
+      next if value.nil?
+
+      Current.public_send("#{key}=", value)
+    end
+  end
+
+  def stash_redirect_params
+    return unless response.redirect?
+
+    redirect_params = params.permit(*Current::PERSISTED_ATTRIBUTES)
+    flash[:redirect_params] = redirect_params.to_h if redirect_params.present?
+  end
```

Comment:
I'm not sure if `flash` is the best mechanism to implement this feature, since `flash` is intended for messages only. Couldn't this be implemented as a query string parameter?

Reply:
We can also consider `session`

Reply:
I don't think it is just for messages, true that this is the most common use:
https://api.rubyonrails.org/v7.2/classes/ActionDispatch/Flash.html

But I considered URLs, Session and Flash. Flash seems the simpler solution.
Happy to discuss in a demo session